File: java/src/main/java/org/tvarley/euler/solutions/Solution016.java
/*
 * Solution for Euler problem 016
 *
 * Power digit sum
 *
 * 2^15 = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.
 *
 * What is the sum of the digits of the number 2^1000?
 *
 * @author tvarley
 * @date 3/8/2026
 */
package org.tvarley.euler.solutions;

import org.tvarley.euler.Solution;
import java.math.BigInteger;

public class Solution016 implements Solution {
  public String solve() {
    BigInteger power = BigInteger.valueOf(2).pow(1000);
    String powerStr = power.toString();
    int sum = 0;
    for (char c : powerStr.toCharArray()) {
      sum += c - '0';
    }
    return Integer.toString(sum);
  }
}

File: javascript/src/euler/solution016.js
// https://projecteuler.net/problem=16

// 2^15 = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.
// What is the sum of the digits of the number 2^1000?
//
// Answer: 1366

module.exports = {
  answer: () => {
    let num = 2n ** 1000n;
    let sum = 0;
    let numStr = num.toString();

    for (let digit of numStr) {
      sum += parseInt(digit);
    }

    return sum;
  }
};